verifier/truststore/truststore.go

```diff
@@ -89,40 +87,35 @@ func (trustStore *x509TrustStore) GetCertificates(ctx context.Context, storeType
 		if err != nil {
 			return nil, fmt.Errorf("error while reading certificates from %q: %w", joinedPath, err)
 		}
-
-		if err := validateCerts(certs, joinedPath); err != nil {
+		if err := ValidateCerts(certs, storeType, namedStore); err != nil {
 			return nil, err
 		}
-
 		certificates = append(certificates, certs...)
 	}
-
 	if len(certificates) < 1 {
 		return nil, fmt.Errorf("trust store %q has no x509 certificates", path)
 	}
-
 	return certificates, nil
 }
 
-func validateCerts(certs []*x509.Certificate, path string) error {
-	// to prevent any trust store misconfigurations, ensure there is at least
-	// one certificate from each file.
+// ValidateCerts ensures certs from trust store are CA certificates or
+// self-signed.
+func ValidateCerts(certs []*x509.Certificate, storeType Type, namedStore string) error {
 	if len(certs) < 1 {
-		return fmt.Errorf("could not parse a certificate from %q, every file in a trust store must have a PEM or DER certificate in it", path)
+		return fmt.Errorf("could not parse a certificate from %s/%s, every file in a trust store must have a PEM or DER certificate in it", storeType, namedStore)
 	}
```

Comment:
The function `func ValidateCerts(certs []*x509.Certificate, storeType Type, namedStore string) error` is not generic enough than previous `func ValidateCertificateChain(certs []*x509.Certificate) error`. Also, other implementation may not read certificates from file so that PEM and DER may not be applicable.